Matrix.java

```diff
@@ -0,0 +1,210 @@
+class Matrix
+{
+    final double[][] a;
+    
+    Matrix(double[][] A)
+    {
+	this(A,0,0);
+    }
+    Matrix(double[][] A, int m, int n)
+    {
+        int rows = A.length;
+        m = rows < m ? rows : m;
```

Comment:
I'm sorry, @whereswaldon how did you manage to display lines 9-10 in your comment? I would have to agree that the functionality defined in the documentation does not match our code. It seems logical to only copy the portions of the array that fit.